### ISSUE TITLE
Enable nested id and parentId properties

### ISSUE THREAD

**Issue by gregoryforel** [philipstanislaus/performant-array-to-tree#20]
Here's my data structure: 
```
    const array = [
        {  data: { id: 'id1', parent: 'idparent' , otherdata...} },
        {  data: { id: 'id2', parent: 'id1' , otherdata...} },
        {  data: { id: 'id3', parent: 'id1' , otherdata...} }
    ]
```

`id` and `parentId` are nested inside `data` in my case, and I'd bet it is not uncommon.

Would you amend your code this way to make this scenario work?

**Add this:**
```
    const getItemNestedProperty = (item: Item, nestedProperty: string): string => {
	return nestedProperty.split('.').reduce((o, i) => o[i], item)
    }
```
**Replace**
```
    const itemId = item[conf.id]
    const parentId = item[conf.parentId]
````
with:
```
    const itemId = getItemNestedProperty(item, conf.id)
    const parentId = getItemNestedProperty(item, conf.parentId)
```

Thanks!

**Comment by philipstanislaus** [comment]
@gregoryforel that's a valuable addition, happy to add it. Would you submit a PR for that change? Many thanks!

**Comment by gregoryforel** [comment]
@philipstanislaus sure, done

**Comment by philipstanislaus** [comment]
Landed in #21. Thanks again for your contribution 💪
